### PR TITLE
Update streamly to the latest commit

### PR DESCRIPTION
**Pull request by adithyaov**

### CODE REVIEW

File: src/Streamly/External/ByteString.hs [hunk 1]
```diff
@@ -13,45 +14,51 @@ where
...
+makeForeignPtr contents (Ptr addr#) =
+    ForeignPtr addr# (arrayToFptrContents contents)
+
 -- | Convert a 'ByteString' to an array of 'Word8'. This function unwraps the
 -- 'ByteString' and wraps it with 'Array' constructors and hence the operation
 -- is performed in constant time.
 {-# INLINE toArray #-}
 toArray :: ByteString -> Array Word8
-toArray (PS fp off len) = Array nfp endPtr
+toArray (PS (ForeignPtr addr# _) _ _)
```
Comment (harendra-kumar) [inline src/Streamly/External/ByteString.hs:42]: Can we use fromForeignPtrUnsafe here? We have that in mutable array module but we can make one for immutable module as well. Also, change that to handle the nullAddr# case. That will also make that routine more robust.

File: src/Streamly/External/ByteString.hs [hunk 2]
```diff
@@ -13,45 +14,51 @@ where
...
+    | Ptr addr# == nullPtr = Array nilArrayContents nullPtr nullPtr
```
Comment (harendra-kumar) [inline src/Streamly/External/ByteString.hs:43]: `addr# == nullAddr#`?

File: src/Streamly/External/ByteString.hs [hunk 3]
```diff
@@ -13,45 +14,51 @@ where
...
+    startPtr = Ptr addr# `plusPtr` off
+    endPtr = startPtr `plusPtr` len
 
 -- | Convert an array of 'Word8' to a 'ByteString'. This function unwraps the
 -- 'Array' and wraps it with 'ByteString' constructors and hence the operation
 -- is performed in constant time.
 {-# INLINE fromArray #-}
 fromArray :: Array Word8 -> ByteString
+fromArray arr
+    | arrStart arr == nullPtr = mempty
```
Comment (harendra-kumar) [inline src/Streamly/External/ByteString.hs]: Why do we need this? `aLen == 0` below should be enough. We never set arrStart to nullPtr anyway.

**Review by harendra-kumar — CHANGES_REQUESTED** [review]

File: .github/workflows/packcheck.yml [hunk 4]
```diff
@@ -95,56 +95,66 @@ jobs:
...
-        command:
-          [ cabal-v2
-          # , stack
-          ]
-        runner:
-          [ ubuntu-latest
-          # , macos-latest
-          ]
+        name:
+          # - 9.2.1
```
Comment (harendra-kumar) [inline .github/workflows/packcheck.yml]: Can we enable this as well?

File: src/Streamly/External/ByteString.hs (hunk 1 above)
Comment (adithyaov) [reply src/Streamly/External/ByteString.hs:42]: We can. I'll add it.

File: src/Streamly/External/ByteString.hs (hunk 3 above)
Comment (adithyaov) [reply src/Streamly/External/ByteString.hs]: > We never set `arrStart` to nullPtr anyway.

We do on an empty bytestring

File: src/Streamly/External/ByteString.hs (hunk 2 above)
Comment (adithyaov) [reply src/Streamly/External/ByteString.hs:43]: How do you compare unlifted types?

**Review by harendra-kumar — APPROVED** [review]

**Pull request merged**